### PR TITLE
Upgrade jasmine, add karma and nodeJS support

### PR DESCRIPTION
**Pull request by hthetiot**

### CODE REVIEW

File: require.js [hunk 1]
```diff
@@ -138,16 +138,17 @@
...
+                var module = getModuleDescriptor(id);
+                module.exports = exports;
+                module.location = URL.resolve(config.location, id);
+                module.directory = URL.resolve(module.location, "./");
+                module.injected = true;
+                module.redirect = void 0;
+                module.mappingRedirect = void 0;
+                // delete module.redirect;
+                // delete module.mappingRedirect;
+            });
```
Comment (kriskowal) [inline require.js]: Why does this need to return a promise? This breaks the guarantee that inject will patch the module synchronously.

Comment (hthetiot) [inline require.js]: @kriskowal  Yes I saw some case it was needed. I will rollback.

Comment (hthetiot) [inline require.js]: That was to fail more gracefully in case of JS error in URL.resolve of getModuleDescriptor basically.

**Comment by hthetiot** [comment]
Only mop fail on travis, otherwise all spec pass.

**Comment by hthetiot** [comment]
@kriskowal I have triple checked for regression and I dont see any. I dont expect hot fix but i would fix master after merge if needed.

I have check with @marchant for merging this has a start for version 17.0.0 in master, see change on package.json version. #ReleaseSoonReleaseOften

**Comment by hthetiot** [comment]
screen shot 2017-03-29 at 9 29 48 am

Merging now.

**Pull request merged**